### PR TITLE
Coil disk cache

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### Description

Fixed the design of login screen and enabled `Coil` disk cache. As `Coil` doesn't have its own disk cache we have to rely on `http` cache. Therefore, added `Cache-Control` header to `OkHttp` interceptor chain.

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

| Before | After |
| --- | --- |
| photo_2020-12-14 11 31 41 | photo_2020-12-14 11 32 03 |

### CODE REVIEW

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt [hunk 1]
```diff
@@ -19,8 +21,19 @@ public class StreamImageLoaderFactory(
             .allowHardware(false)
             .crossfade(true)
             .okHttpClient {
+                val cacheControlInterceptor = Interceptor { chain ->
+                    chain.proceed(chain.request())
+                        .newBuilder()
+                        .header("Cache-Control", "max-age=31536000,public")
```
Comment (zsmb13) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt:27]: > 31536000

This is one year, right? Isn't that a bit long for caching images?

Comment (bychkovdmitry) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt:27]: Agree, what's the reasonable time span? One hour?

Comment (bychkovdmitry) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt:27]: At the same time, all this config looks weird, we should allow customers to take control over image loading
chatUI.setImageLoader(StreamCoilImageLoader(clientsCoilInstance))
chatUI.setImageLoader(StreamGlideImageLoader(clientsGlideInstance))
chatUI.setImageLoader(StreamPicassoImageLoader(clientsPicassoInstance))
chatUI.setImageLoader(object: StreamImageLoader() { // completely custom image loader implemented by client  })

Comment (bychkovdmitry) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt]: Changed cache expiration time to 1 hour

Comment (zsmb13) [reply stream-chat-android/src/main/java/com/getstream/sdk/chat/coil/StreamImageLoaderFactory.kt]: >we should allow customers to take control over image loading

That can go on the backlog 😄 

>Changed cache expiration time to 1 hour

Sounds good for a start, we can revisit it later if we have to

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**